core/electron-manager/src/backend/ElectronWindowState.ts

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { NativeAppStorage } from "@bentley/imodeljs-backend";
+import { StorageValue } from "@bentley/imodeljs-common";
+import { BrowserWindow } from "electron";
+
+// cSpell:ignore unmaximize
+
+/** Maintains main Electron window state for the application.
+ * @beta
+ */
+export class ElectronWindowState {
+  private static _xSettingName = "electron-window-x";
+  private static _ySettingName = "electron-window-y";
+  private static _widthSettingName = "electron-window-width";
+  private static _heightSettingName = "electron-window-height";
+  private static _maximizedSettingName = "electron-window-maximized";
+
+  /** Constructor for ElectronWindowState.
+   * @param storageName       Name used for NativeAppStorage when saving size, position and maximized state
+   * @param defaultWidth      Default width of window
+   * @param defaultHeight     Default height of window
+   * @param defaultMaximized  Default maximized state of window
+   */
+  constructor(public storageName: string, public defaultWidth: number, public defaultHeight: number, public defaultMaximized: boolean = false) {

```

Comment:
maybe we should make up a storage that all electron apps use instead of everybody creating a new one.

Reply:
Updated